scripts/build_jacocorunner/build_jacocorunner.sh

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env bash
+#
+# Script to build custom version of `JacocoCoverage_jarjar_deploy.jar` from Jacoco and Bazel repositories.
+#
+# The default `JacocoCoverage_jarjar_deploy.jar` has some issues:
+#
+# 1. Bazel uses Jacoco 0.8.3 that has poor Scala support, including a bug that filters out Scala lambdas on Scala >=2.12
+#
+#    Bug report:
+#    https://github.com/bazelbuild/rules_scala/issues/1056
+#    https://github.com/bazelbuild/bazel/issues/11674
+#
+#    Backported fix from Jacoco 0.8.5 to Jacoco 0.8.3:
+#    https://github.com/gergelyfabian/jacoco/tree/0.8.3-scala-lambda-fix
+#
+# 2. Bazel ignores Jacoco's filtering for branch coverage metrics:
+#
+#    Bug report:
+#    https://github.com/bazelbuild/bazel/issues/12696
+#
+#    Proposed fix:
+#    https://github.com/gergelyfabian/bazel/tree/branch_coverage_respect_jacoco_filtering
+#
+# 3. Scala support on newer Jacoco versions (including 0.8.5) is still lacking some functionality
+#
+#    E.g. a lot of generated methods for case classes, lazy vals or other Scala features are causing falsely missed branches in branch coverage.
+#
+#    Proposed changes in:
+#    https://github.com/gergelyfabian/jacoco/tree/scala
+#
+#    Backported to 0.8.3 (to be usable with current Bazel):
+#    https://github.com/gergelyfabian/jacoco/tree/0.8.3-scala
+#
+#    These branches also include the Scala 2.12 lambda coverage fix.
+#
+# You can use this script to build a custom version of `JacocoCoverage_jarjar_deploy.jar`, including any fixes from the above list you wish
+# and then provide the built jar as a parameter of `java_toolchain` and/or `scala_toolchain` to use the changed behavior for coverage.
+#
+# Choose the fixes from the above list by configuring the used branches for Bazel and Jacoco repos below.
+
+set -e
+
+source_path=$(readlink -f $(dirname ${BASH_SOURCE[0]}))
+
+jacoco_repo=$HOME/Work/jacoco
```

Comment:
Does it assume that users have `Work` in their home folder? I would prefer that script runs in an isolated directory (probably a random one under `tmp`). This way it would be easier to get this script running on other user's machines. And potentially won't accidentally overwrite users work.

Reply:
I'd vote for creating a deterministic one under `/tmp`, the reason is to make potential re-checkout and rebuild faster (Bazel cache, etc.), and also to be able to output the built artifact in that directory, instead of to `HOME` (for a random dir I'd need to remove that, but then where to write the built artifact?).

Reply:
I've pushed this change.